files/cache.py

```diff
@@ -42,7 +42,9 @@ def addGroups(self, username:str, matchedGroups:list):
 		'''
 		if username in self.cache:
 			self.logs.info({'message':'Adding groups to the cache.', 'username': username, 'matchedGroups': ','.join(matchedGroups)})
-			self.cache[username]['matchedGroups'] = matchedGroups
+			for group in matchedGroups:
+				if not (group in self.cache[username]['matchedGroups']):
+					self.cache[username]['matchedGroups'].append(group)
```

Comment:
there is a one line way of combining lists without duplicates
`self.cache[username]['matchedGroups'] = list(set(self.cache[username]['matchedGroups'] + matchedGroups))`

Reply:
The approach was modified with the suggestion of @marianobilli